lib/perl/Genome/InstrumentData/AlignmentResult.pm

```diff
@@ -1661,8 +1660,14 @@ sub revivified_alignment_bam_file_paths {
         return @bams;
     }
 
-    my $revivified_bam = File::Spec->join($p{disk_allocation}->absolute_path, 'all_sequences.bam');
-    my $merged_bam    = $self->get_merged_bam_to_revivify_per_lane_bam;
+    my $temp_allocation = $self->_get_temp_allocation($self->output_dir);
+    UR::Context->process->add_observer(
```

Comment:
My gut feeling is that this is not going to work well.  I think the `commit` aspect is "post-commit" and you need to commit the deletion of the allocation as well.  You might be able to move it to `pre-commit` (I think that exists).  However, a guard object that destroys the allocation when it goes out of scope might be better overall.